usefulUtilities/hiFiCalendar/entityServerScripts/hiFiCalendarServer.js

```diff
@@ -8,119 +8,348 @@
+            } catch (e) {
+                console.log(e, "Could not parse userData");
+                return;
+            }
+            that.roomScheduleID = that.userData.roomScheduleID; 
+            that.roomOccupantListID = that.userData.roomOccupantListID; 
+        } else {
+            console.log("Please enter appropriate userData to enable functionality of this server script.");
+            return;
+        }
+
+        if (that.entityID === that.roomScheduleID) {
+            that.sentAlready = false;
+            that.token = that.userData.token;
+            that.expireTime = that.userData.expireTime;
+            that.timezoneOffset = that.userData.timezoneOffset;
+            that.timezoneName = that.userData.timezoneName;
+            that.calendarID = that.userData.roomCalendarAddress;
+            that.roomColorID = that.userData.roomColorID;
+            that.roomColorOccupantsID = that.userData.roomColorOccupantsID; 
+            that.roomEntityIDs = [
+                that.roomScheduleID,
+                that.roomColorID,
+                that.roomColorOccupantsID,
+                that.roomOccupantListID
+            ];                
+            that.room = {};
             that.room = {
-                "id": that.entityProperties.id, 
-                "name": that.entityProperties.name,
-                "eventList": [],
-                "scheduleEntityID": that.entityProperties.parentID
+                "eventList": []
             };
-            Entities.editEntity(that.room.scheduleEntityID, {text: ""});
-            that.updateSignColor(that.room.id, [that.room.name, "GREEN"]);
-        } else {
+            that.request = Script.require('https://hifi-content.s3.amazonaws.com/brosche/dev/googleCalendar/resources/modules/request.js').request;
+            that.clearEventList(that.entityID);
+            that.updateSignColor(that.roomEntityIDs[1], ["GREEN"]);
+            that.updateSignColor(that.roomEntityIDs[2], ["GREEN"]);
+            that.googleRequest(that.token);
+        } else if (that.entityID === that.roomOccupantListID) {
+            that.room = {};
             that.room = {
-                "id": that.entityProperties.id, 
                 "occupants": []
             };
-            return;
+            Entities.editEntity(that.entityID, {
+                text: 'loading'
+            });
+        }
+    };
+
+
+    this.refreshToken = function(id) {
+        if (that.entityID === id) {
+            if (that.interval) {
+                console.log("CLEARING INTERVAL");
+                Script.clearInterval(that.interval);
+                that.interval = false;
+            }
+            that.entityProperties = Entities.getEntityProperties(that.entityID, ['userData']);
+            if (that.entityProperties.userData.length !== 0) {
+                try {
+                    that.userData = JSON.parse(that.entityProperties.userData);
```

Comment:
why can't you do `var userData` instead of `that.userData` ? i don't think you want to be making `userData` an object inside `that`, do you?

Reply:
see above response.